x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/PyTorchResultProcessor.java

```diff
@@ -56,6 +56,20 @@ public void process(NativePyTorchProcess process) {
             if (isStopping == false) {
                 logger.error(new ParameterizedMessage("[{}] Error processing results", deploymentId), e);
             }
+            // If we are stopping, or the process is no longer alive, notify all the pendingResults listeners
+            // Otherwise, we don't particularly know which result failed
+            if (isStopping || (process.isProcessAliveAfterWaiting() == false)) {
+                pendingResults.forEach((id, pendingResults) -> {
+                    pendingResults.result = new PyTorchResult(
+                        id,
+                        null,
+                        null,
+                        isStopping ?
+                            "inference canceled as process is stopping" :
+                            "inference native process died unexpectedly with failure [" + e.getMessage() + "]");
+                    pendingResults.latch.countDown();
+                });
+            }
```

Comment:
Add `pendingResults.clear();` just to be nice

Reply:
we could `clear()` here, the `PendingResult pendingResult = pendingResults.computeIfAbsent(requestId, k -> new PendingResult());` has a local reference, so there isn't really a race condition on removing here vs the `waitForResult` method